dlib/test/image.cpp

```diff
@@ -1987,6 +1987,222 @@ namespace
         }
     }
 
+    template<typename interpolation_type = interpolate_bilinear>
+    void test_resize_image_with_interpolation()
+    {
+        {
+            matrix<unsigned char> img_s(2, 2);
+            matrix<unsigned char> img_d(3, 3);
+
+            img_s(0, 0) = 0;
+            img_s(0, 1) = 100;
+            img_s(1, 0) = 100;
+            img_s(1, 1) = 100;
+
+            resize_image(img_s, img_d, interpolation_type());
+            DLIB_TEST((img_d(0, 0) == 0));
+            DLIB_TEST((img_d(0, 1) == 50));
+            DLIB_TEST((img_d(1, 2) == 100));
+            DLIB_TEST((img_d(2, 2) == 100));
```

Comment:
The dlib image has an operator== that compares for equality.  So you can just do:
`DLIB_TEST(img_d == img_s)`.  That will make the intention of the test more clear.

Reply:
Hmm, bool operator== (const matrix_exp<EXP1>& m1, const matrix_exp<EXP2>& m2) uses operator!= on pixel which is missing. Should I add it (preferable?), or correct to !(p1 == p2)?

Reply:
Went for `operator!=` as for me makes sense to have it anyway if `operator==` is there.